source/simulator/stokes_matrix_free.cc

```diff
@@ -1322,7 +1322,6 @@ namespace aspect
                                sim.pressure_scaling,
                                sim.triangulation,
                                dof_handler_projection);
```

Comment:
the mass matrix diagonal is needed for the preconditioner. It seems more appropriate in the `build_preconditioner()` function.